### DISCUSSION TITLE
[Feature request] add Maintainer role to achievements

### DISCUSSION THREAD

**Discussion by suspect15** [RetroAchievements/RAWeb#2720] [Ideas]
It is not uncommon for older achievements/sets from inactive developers to need substantial logic rewriting.  In some cases where entire sets were recoded, limited reauthorship has been granted by DevCompliance.  A reauthorship essentially replaces an original author with a new one, yielding all credit and unlock awards to the new author.  This in a sense erases the history of the achievement as sometimes substantial numbers of unlocks are awarded to someone who had nothing to do with an achievement when potentially thousands of unlocks were accrued.

At its core, RetroAchievements is largely a historical archive of achievements that players have earned throughout their experience with the project.  Part of RA's history are the records and statistics of its contributors.  There isn't currently a way to give credit to an achievement's original author as well as a developer who may have recoded it to ensure future stability.

An idea suggested during a discussion on this topic was a maintainer role that could coexist along side author credit for achievements.  This would allow for both past and present/future developers to receive credit for their contributions to a set.  If built and implemented, Developer Code of Conduct guidelines would need to be written to determine the necessary threshold for being designated an achievement's maintainer.

Below is a set of initial of requirements/desires.  There are likely more aspects to consider.

1. A way to designate a current dev as an achievement maintainer which is the responsible party for addressing any future tickets on an achievement
2. Give future unlock award credit to the developer once designated as maintainer
3. Preserve unlock award credit for the original author up until the time when a maintainer is designated
4. Maintainer acknowledgement on set pages
5. Both maintainer and author acknowledgement under individual achievements

**Comment by redwizard42** [comment]
I think:
- Current maintainer should be listed first followed by any past maintainers (oldest last), followed by original author. Anyone contacting the responsible party will likely contact the first dev listed.
- Is maintainer per set? Can/should it be per achievement, which may more easily allow for multiple maintainers?
- Upon becoming a maintainer, the maintainer should get assigned any open tickets on the maintained [set/achievement]. New tickets will then also be assigned to the maintainer 
- Unlocks get attributed to the active maintainer, so OG author gets unlocks up until the date a maintainer becomes active after which the maintainer gains the unlocks. If a new maintainer takes over, unlocks from that point forward to to them, etc.

**Reply by suspect15** [reply]
This is better stated than mine.

**Comment by wescopeland** [comment]
Implemented in https://github.com/RetroAchievements/RAWeb/pull/3452.
